File: client-java/src/main/java/io/maxi/examples/client/PlayerStats.java
package io.maxi.examples.client;

import io.maxi.annotation.MaxiField;
import io.maxi.annotation.MaxiType;

/**
 * PlayerStats DTO — mirrors the S:PlayerStats type in shared/sports.mxs.
 */
@MaxiType(alias = "S", name = "PlayerStats")
public class PlayerStats {

    @MaxiField(typeExpr = "P")
    public Player player;

    @MaxiField(typeExpr = "int", defaultValue = "0")
    public int goals;

    @MaxiField(typeExpr = "int", defaultValue = "0")
    public int assists;

    @MaxiField(typeExpr = "int", defaultValue = "0")
    public int minutesPlayed;

    public PlayerStats() {}
}

File: client-php/model/PlayerStats.php
<?php

declare(strict_types=1);

use Maxi\Attribute\MaxiField;
use Maxi\Attribute\MaxiType;

/**
 * PlayerStats DTO — mirrors the S:PlayerStats type in shared/sports.mxs.
 */
#[MaxiType(alias: 'S', name: 'PlayerStats')]
class PlayerStats
{
    public function __construct(
        #[MaxiField(typeExpr: 'P')]
        public ?Player $player = null,

        #[MaxiField(typeExpr: 'int', defaultValue: 0)]
        public int $goals = 0,

        #[MaxiField(typeExpr: 'int', defaultValue: 0)]
        public int $assists = 0,

        #[MaxiField(typeExpr: 'int', defaultValue: 0)]
        public int $minutesPlayed = 0,
    ) {}
}
